### ISSUE TITLE
Wrong element is highlighted

### ISSUE THREAD

**Issue by sguptatw** [getgauge/taiko#145]
**Expected behavior**
Highlight API should highlight based on the selector

**Actual behavior**
Highlight is not considering the selector details

```
const { openBrowser, goto, link, click, highlight, above } = require('taiko');

(async () => {
    try {
        await openBrowser();
        await goto("google.com");
        await click(link({"title":"Google apps"}));
        await highlight(contains("Gmail"),below(link("Gmail")))
        await screenshot()
    } catch (e) {
        console.error(e);
    } finally {
        closeBrowser();
    }
})();
```
![narrowsearch](https://user-images.githubusercontent.com/6310197/45202730-8bcbdf80-b297-11e8-916d-a5a0eb4efd0a.png)

**Another example**
```
const { openBrowser, goto, scrollTo, click, button, link } = require('taiko');

(async () => {
    try {
        await openBrowser({headless:false});
        await goto("http://demos.jquerymobile.com/1.4.5/popup/");
        await scrollTo("sign in");
        await click(link("Sign in"));
        await highlight(link("Sign in"));
        await click(button("Sign in"));
        await highlight(button("Sign in"));
        await waitFor("10000")
    } catch (e) {
        console.error(e);
    } finally {
        closeBrowser();
    }
})();
```
**Version**
```
24451f49f67f9376f3fb6b1734c4f6c0e6332a09
```
Blocks #73

**Comment by NivedhaSenthil** [comment]
This is independent of #73. This is happening because highlight API does not have support for proximity selectors which has to be added. And the second script is working as expected.

**Comment by sguptatw** [comment]
This is not directly related. But IMO is important for #73, 'cause currently wrong element is highlighted during observe or REPL. This will give user a false perception of what is going to be clicked. The user may choose to change/retain the script based on a wrong detail given by highlight.

**Comment by negiDharmendra** [comment]
@sguptatw @NivedhaSenthil what would be the use case of adding support for proximity selector.
A user can always pass the final selector.
**Example:**
> highlight( contains( "Gmail" ), below( link( "Gmail" ) ) )
>  instead of 
> highlight( contains( "Gmail", below( link( "Gmail" ) ) )

**Comment by NivedhaSenthil** [comment]
@negiDharmendra  no specific use case I could think of, it can always be done like mentioned above.
Just that we might have to maintain consistency among actions as `click` api currently supports it.

**Comment by NivedhaSenthil** [comment]
Fix available in 86d4f9f
